src/tribler-common/tribler_common/sentry_reporter/tests/test_sentry_reporter.py

```diff
@@ -0,0 +1,188 @@
+            self._allow_sending = value
+
+    return MockReporter()
+
+
+def test_init(reporter):
+    assert reporter.init('')
+
+
+def test_states():
+    instance1 = SentryReporter()
+    instance2 = SentryReporter()
+    assert instance1 != instance2
+    assert instance1.get_allow_sending() == instance2.get_allow_sending()
+
+    instance1.allow_sending(not instance2.get_allow_sending())
+    assert instance1.get_allow_sending() == instance2.get_allow_sending()
+
+
+def test_allow_sending(reporter):
+    reporter.allow_sending(True)
+    assert reporter.get_allow_sending() is True
+
+    reporter.allow_sending(False)
+    assert reporter.get_allow_sending() is False
+
+    reporter.allow_sending(True, 'info')
+    assert reporter.get_allow_sending() is True
+
+
+def test_set_user(reporter):
+    assert reporter.set_user('some_id'.encode('utf-8')) == \
+           {'id': 'db69fe66ec6b6b013c2f7d271ce17cae',
+            'username': 'Wanda Brown'}
+
+    assert reporter.set_user(b'11111100100') == \
+           {'id': '91f900f528d5580581197c2c6a4adbbc',
+            'username': 'Jennifer Herrera'}
+
+
+def test_allow_sentry_reports(mock_reporter):
+    # test base logic
+    mock_reporter.allow_sending(None)
+
+    assert mock_reporter.get_allow_sending() is None
+    with AllowSentryReports(reporter=mock_reporter):
+        assert mock_reporter.get_allow_sending() is True
+
+    assert mock_reporter.get_allow_sending() is None
+
+    with AllowSentryReports(value=False, reporter=mock_reporter):
+        assert mock_reporter.get_allow_sending() is False
+
+    assert mock_reporter.get_allow_sending() is None
+
+    # test that a correct reporter returned
+    with AllowSentryReports(reporter=mock_reporter) as reporter:
+        assert mock_reporter.get_allow_sending() is True
+        assert reporter == mock_reporter
+
+    # test default initialisation
+    with AllowSentryReports() as reporter:
+        assert reporter
+
+
+def test_send(reporter):
+    assert reporter.send(None, None, None) is None
+
+    # test return defaults
+    assert reporter.send({}, None, None) == {
+        'contexts': {'browser': {'name': 'Tribler', 'version': None},
+                     'reporter': {'_stacktrace': [],
+                                  'comments': None,
+                                  'os.environ': {},
+                                  'sysinfo': None}},
+        'tags': {'machine': None,
+                 'os': None,
+                 'platform': None,
+                 'platform.details': None,
+                 'version': None}}
+
+    # test post_data
+    post_data = {
+        "version": '0.0.0',
+        "machine": 'x86_64',
+        "os": 'posix',
+        "timestamp": 42,
+        "sysinfo": '',
+        "comments": 'comment',
+        "stack": 'some\nstack',
+    }
+
+    assert reporter.send({'a': 'b'}, post_data, None) == {
+        'a': 'b',
+        'contexts': {'browser': {'name': 'Tribler', 'version': '0.0.0'},
```

Comment:
This stuff could really use some `black` formatting.

Reply:
Why should we use black formatting?

Reply:
https://github.com/Tribler/tribler/issues/4104

Short answer: because we had a thorough discussion about code autoformatting and decided that we will use it in the GUI and in some parts of the codebase.

Actually, I should have already added `pre-commit` checks to Jenkins jobs to check the conformance. I'll probably do it this weekend.

Reply:
@ichorid no, please! 

Now we have:
1. pylint check
2. flake8 check
3. sonarcloud
4. deepcode-ci-bot

And you want to add a mandatory `pre-commit` check 🤦 

We need to decrease the count of checking tools, not increase.

Your citate:
> After that, every developer's work will become much easier, as instead of fixing pylint errors by hand the developer will only have to hit Ctl-Alt-L once in their PyCharm

That's not true at all! I spent a lot of my time in an effort to please the numerous checkers *after* Ctl-Alt-L. 
It is boring and annoying.

PS In the issue, you gave me I see no consensus on `Black`

Reply:
>That's not true at all! I spent a lot of my time in an effort to please the numerous checkers after Ctl-Alt-L.
It is boring and annoying.

I had the same problem before I embraced `pre-commit`, never had to appease import checkers and linters since that moment: `black` and other `pre-commit`-invoked tools do all the job for me. Try it out, you'll like it! :wink:

Reply:
In the issue, the consensus was: for those areas of code that are marked as `black`-formatted, we use `black`. Developers may add *the code they own* to `black` areas if they wish so. I own the GUI code (together with @devos50) and the MetadataStore code. `tribler-common` was created by me during refactoring, so it is owned by me now. If you wish, you 
can exclude the `sentry`-related code from `black`-area in `tribler-common`- that's your code after all.

Regardless of that, `pre-commit` also **performs automatic import sorting for the whole codebase** and runs lint-checks locally (for the programmer's convenience).

Reply:
`pre-commit` formating has been applied